examples/hardhat/scripts/run-test.ts

```diff
@@ -0,0 +1,49 @@
+import * as childProcess from 'child_process'
+import * as dotenv from 'dotenv'
+import { resolve } from 'path'
+import * as waitOn from 'wait-on'
+
+dotenv.config()
+
+// @ts-ignore
+import _spawn = require('cross-spawn-with-kill')
+
+const spawn = _spawn as typeof import('child_process').spawn
+
+async function main() {
+  let port = process.env.HH_NODE_PORT
+  if (!port) {
+    // eslint-disable-next-line no-eval
+    port = String(await eval("import('get-port')").then((m: typeof import('get-port')) => m.default()))
+    process.env.HH_NODE_PORT = port
+  }
+
+  console.log(`Starting hardhat node on port ${port}`)
+
+  const node = spawn('hardhat', ['node', '--port', port], { stdio: 'ignore' })
+  node.on('error', (err) => {
+    console.error(err)
+    process.exit(1)
+  })
+  await waitOn({
+    resources: [`http://localhost:${port}/`],
+  })
+
+  childProcess.execSync('yarn eth-sdk', { stdio: 'inherit' })
+
+  console.log('Generated SDK with eth-sdk')
+
+  if (process.env.ARCHIVE_NODE_RPC_URL) {
+    console.log('Running ./scripts/exploit.ts')
+    childProcess.execSync('hardhat run ./scripts/exploit.ts', { cwd: resolve(__dirname, '..'), stdio: 'inherit' })
+  } else {
+    console.log('Skipping examples/hardhat test because process.env.ARCHIVE_NODE_RPC_URL is not set')
+    if (process.env.CI) {
+      process.exit(1)
+    }
+  }
+
+  node.kill()
+}
+
+void main()
```

Comment:
This was a bit annoying to be fair. Should we request a feature from Hardhat to make it simpler?

```js
hardhat.runWithNode((rpcUrl) => { /* ... */ })
```

```sh
hardhat node run "yarn eth-sdk"
```

🤔

Reply:
We could probably make a hardhat plugin for eth-sdk too, just to achieve that, but I think the case can be generalized and we're not the only tool that needs it.

Reply:
Good idea to make an issue in hardhat repo 👍